nemo/deploy/nlp/trtllm_pytorch_deployable.py

```diff
@@ -0,0 +1,209 @@
+        )
+
+    def generate(
+        self,
+        prompts: List[str],
+        max_length: int = 256,
+        temperature: Optional[float] = None,
+        top_k: Optional[int] = None,
+        top_p: Optional[float] = None,
+        **kwargs,
+    ) -> List[str]:
+        """Generate text based on the provided input prompts.
+
+        This method processes input prompts through the loaded model and
+        generates text according to the specified parameters.
+
+        Args:
+            prompts: List of input prompts
+            max_length: Maximum number of tokens to generate. Defaults to 256.
+            temperature: Sampling temperature. Defaults to None.
+            top_k: Number of highest probability tokens to consider. Defaults to None.
+            top_p: Cumulative probability threshold for token sampling. Defaults to None.
+            **kwargs: Additional keyword arguments to sampling params.
+
+        Returns:
+            List[str]: A list of generated texts, one for each input prompt.
+
+        Raises:
+            RuntimeError: If the model is not initialized.
+        """
+        if not self.model:
+            raise RuntimeError("Model is not initialized")
+
+        sampling_params = SamplingParams(
+            max_tokens=max_length,
+            temperature=temperature,
+            top_k=top_k,
+            top_p=top_p,
+            **kwargs,
+        )
+
+        outputs = self.model.generate(
+            inputs=prompts,
+            sampling_params=sampling_params,
+        )
+
+        return [output.outputs[0].text for output in outputs]
+
+    def generate_other_ranks(self):
+        """
+        Generate function for ranks other than the rank 0.
+        """
+        while True:
+            message = torch.empty(1, dtype=torch.long, device="cuda")
+            torch.distributed.broadcast(message, src=0)
+            if message == 0:
+                prompts = broadcast_list(data=[None], src=0)
+                temperature, top_k, top_p, max_length = broadcast_list(data=[None], src=0)
+
+                self.generate(
+                    prompts=prompts,
+                    temperature=temperature,
+                    top_k=top_k,
+                    top_p=top_p,
+                    max_length=max_length,
+                )
+            else:
+                return
+
+    @property
+    def get_triton_input(self):
+        inputs = (
+            Tensor(name="prompts", shape=(-1,), dtype=bytes),
+            Tensor(name="max_length", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="max_batch_size", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="top_k", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="top_p", shape=(-1,), dtype=np.single, optional=True),
+            Tensor(name="temperature", shape=(-1,), dtype=np.single, optional=True),
+            Tensor(name="random_seed", shape=(-1,), dtype=np.int_, optional=True),
```

Comment:
We don't use `"random_seed"` in `triton_infer_fn`.

Is it used anywhere at all?